### PR TITLE
Broadcast FILE-UPDATE

### PR DESCRIPTION
**Pull request by stulees**
- Broadcasting FILE-UPDATE with status "CURRENT" after successfully writing file to disk
- Cleaning up and initializing `download` and `cache` folders
- Integrate logging
- Log critical errors to BQ
- Add the proxy agent to HTTP requests (signed url request and GCS file request)

### CODE REVIEW

File: src/files/file.js [hunk 1]
```diff
@@ -3,14 +3,17 @@ const broadcastIPC = require("../messaging/broadcast-ipc.js");
...
+  const proxy = commonConfig.getProxyAgents();
   const options = {
     uri: signedURL,
     timeout: config.secondMillis * twoMinTimeout,
-    resolveWithFullResponse: true
+    resolveWithFullResponse: true,
+    proxy: proxy.httpsAgent || proxy.httpAgent || null
   };
 
   return request.get(options);
```
Comment (tejohnso) [inline src/files/file.js]: Is this handling retries?

File: src/index.js [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
+const externalLogger = require("common-display-module/external-logger")(config.bqProjectName, config.bqDatasetName, config.bqFailedEntryFile);
+const modulePath = commonConfig.getModulePath(config.moduleName);
 
-database.start()
+global.log = require("rise-common-electron").logger(preventBQLog ? null : externalLogger, modulePath, config.moduleName);
+
+const initialize = () => {
+  return commonConfig.getDisplayId()
+    .then(displayId=>{
+      log.resetLogFiles(Math.pow(10, 5)); // eslint-disable-line no-magic-numbers
```
Comment (tejohnso) [inline src/index.js]: For this one I'd go with a variable for clarity.  I'm guessing it's 100,000 bytes max file size for the failed log file and they're truncated if they're larger than that?

File: src/messaging/messaging.js [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
 
 const handleWatch = (message) => {
   return watch.process(message)
     .catch((err) => {
-      console.log(err);
+      log.error({
+        event_details: JSON.stringify(err),
+        version: commonConfig.getModuleVersion()
+      }, null, config.bqTableName);
     });
```
Comment (santiagonoguez) [inline src/messaging/messaging.js]: as you're repeating this log.error() code 4 times in this file alone, wouldn't it make sense to wrap it in a function ?

File: src/files/file.js (hunk 1 above)
Comment (stulees) [reply src/files/file.js]: @tejohnso Not at the moment. How many retries do you think we should do?

Comment (stulees) [reply src/files/file.js]: @tejohnso Latest commits address retries, please review again, thanks

File: src/index.js (hunk 2 above)
Comment (stulees) [reply src/index.js]: @tejohnso Latest commits address this, thanks

File: src/messaging/messaging.js (hunk 3 above)
Comment (stulees) [reply src/messaging/messaging.js]: @santiagonoguez Created a function to wrap those repetitive logs calls in messaging file. Thanks

**Pull request merged**